File: algorithms/cpp/480.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

class Solution {
public:
    vector<double> medianSlidingWindow(vector<int> &nums, int k) {
        multiset<int> window(nums.begin(), nums.begin()+k);
        auto mid = next(window.begin(), k/2);

        vector<double> medians;
        for (int i = k; ; i++) {
            medians.push_back(((double)*mid + *prev(mid, 1-k%2)) / 2.0);

            if (i == nums.size())
                break;

            window.insert(nums[i]);

            if (nums[i] < *mid)
                mid--;

            if (nums[i-k] <= *mid)
                mid++;

            window.erase(window.lower_bound(nums[i-k]));
        }

        return medians;
    }
};

int main() {
    return 0;
}

File: algorithms/python/480.py
from bisect import bisect_left, insort
class Solution:
    def median(self, nums, k):
        return (nums[k//2] + nums[k//2-(1-k%2)]) / 2

    def medianSlidingWindow(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[float]
        """
        window, rls = sorted(nums[:k]), []
        for i, n in enumerate(nums[k:], k):
            rls.append(self.median(window, k))
            window.pop(bisect_left(window, nums[i-k]))
            insort(window, n)
        rls.append(self.median(window, k))
        return rls
